### PR TITLE
Add ruby runtime, ruby files, and hook

### PR DESCRIPTION
**Pull request by David3141**
This seems to be the easiest way of handling multiple runtime options.

### CODE REVIEW

File: {{cookiecutter.project_slug}}/{{cookiecutter.function_slug}}/Gemfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+source 'https://rubygems.org'
+gem 'aws-record', '~> 2'
```
Comment (timoschilling) [inline {{cookiecutter.project_slug}}/{{cookiecutter.function_slug}}/Gemfile]: please remove `aws-record`

Comment (lagr) [reply {{cookiecutter.project_slug}}/{{cookiecutter.function_slug}}/Gemfile]: Agree, unless we need DynamoDB for something.

**Comment by lagr** [comment]
Should this not be a `sam-ruby-template` ?

**Comment by timoschilling** [comment]
Yes we should rename the repo

**Comment by David3141** [comment]
Wanted to do the renaming afterwards

**Pull request merged**